### PR TITLE
Rename refcount.h to zfs_refcount.h

### PR DESCRIPTION
**Pull request by mattmacy**
Signed-off-by: Matt Macy <mmacy@FreeBSD.org>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/os/freebsd/zfs/zfs_vnops.c [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #include <sys/sysmacros.h>
 #include <sys/resource.h>
 #include <sys/vfs.h>
+#include <sys/endian.h>
```
Comment (behlendorf) [inline module/os/freebsd/zfs/zfs_vnops.c:39]: Did you mean to include this in the PR?

File: module/os/freebsd/spl/spl_vfs.c [hunk 2]
```diff
@@ -39,6 +39,7 @@ __FBSDID("$FreeBSD$");
 
 #include <sys/mutex.h>
 #include <sys/vnode.h>
+#include <sys/taskq.h>
```
Comment (behlendorf) [inline module/os/freebsd/spl/spl_vfs.c:42]: Did you mean to include this in the PR?

File: module/os/freebsd/zfs/zfs_vnops.c (hunk 1 above)
Comment (mattmacy) [reply module/os/freebsd/zfs/zfs_vnops.c:39]: It was implicitly included before. Won't build without.

File: module/os/freebsd/spl/spl_vfs.c (hunk 2 above)
Comment (mattmacy) [reply module/os/freebsd/spl/spl_vfs.c:42]: It was implicitly included before. Won't build without.

**Comment by mattmacy** [comment]
This implicitly relies on the endian changes. I will rebase when  #10621 goes in.

**Comment by freqlabs** [comment]
Something is definitely wrong with this on FreeBSD.

File: include/os/freebsd/spl/sys/ccompile.h [hunk 3]
```diff
@@ -119,7 +119,7 @@ extern "C" {
 #define	__CONST			__sun_attr__((__const__))
 #define	__PURE			__sun_attr__((__pure__))
 
-#ifdef INVARIANTS
+#if defined(INVARIANTS) && !defined(ZFS_DEBUG)
```
Comment (freqlabs) [inline include/os/freebsd/spl/sys/ccompile.h:122]: I have this change in my PR at #10634 so whichever of these goes in first, the other can be rebased.

**Pull request merged**